consensus/core/src/core.rs

```diff
@@ -669,40 +735,48 @@ impl Core {
             }
             assert!(commits_until_update > 0);
 
-            // TODO: limit commits by commits_until_update, which may be needed when leader schedule length
-            // is reduced.
-            let decided_leaders = self.committer.try_decide(self.last_decided_leader);
+            // Always try to process the synced commits first
+            let mut decided_leaders = self
+                .committer
+                .try_decide_synced(&mut synced_commits, commits_until_update);
+
+            // If the synced `decided_leaders` is empty then try to run the decision rule.
+            if decided_leaders.is_empty() {
+                // TODO: limit commits by commits_until_update, which may be needed when leader schedule length is reduced.
+                decided_leaders = self.committer.try_decide(self.last_decided_leader);
```

Comment:
Something to note here; we could avoid even linearizing the DAG and just produce the necessary outputs, persist etc. However right now:
* uitlising the same path saves us from having to specially handle the storing of committed sub dags, move GC rounds etc
* provides some extra layer of confirmation as we can - and we do in the end - make sure that the same sub dag is produced with the current state